src/modules/RA/DataLayer/components/tabs/StyleTab/Style/ColorStyleField.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+
+import { makeStyles } from '@material-ui/core/styles';
+import { Field, useField } from 'react-final-form';
+import { SelectInput, RadioButtonGroupInput, BooleanInput, useTranslate } from 'react-admin';
+import randomColor from 'randomcolor';
+
+import { fieldTypes } from '../../../../../DataSource';
+
+import Condition from '../../../../../../../components/react-admin/Condition';
+
+import ColorPicker from '../../../../../../../components/react-admin/ColorPicker';
+import GraduateValue from './GraduateValue';
+import CategorizeValue from './CategorizeValue';
+
+import ColorListField from './ColorListField';
+
+import styles from './styles';
+
+const useStyles = makeStyles(styles);
+
+const ColorStyleField = ({ path, fields, getValuesOfProperty }) => {
+  const classes = useStyles();
+  const translate = useTranslate();
+
+  const [defaultSeed] = React.useState(Math.floor((Math.random() * 100000) + 1));
+  const [defaultValue] = React.useState([randomColor()]);
+
+  const { input: { value: type } } = useField(`${path}.type`);
+
+  if (type === 'none') {
+    return null;
+  }
+
+  return (
+    <div className={classes.styleField}>
+      <Condition when={`${path}.type`} is="fixed">
+        <Field
+          name={`${path}.value`}
+          defaultValue={randomColor({ seed: defaultSeed })}
+        >{({ input: { onChange, value } }) => (
+          <ColorPicker
+            value={value || '#cccccc'}
+            onChange={onChange}
+          />
+        )}
+        </Field>
+      </Condition>
+
+      <Condition when={`${path}.type`} is="variable">
+
+        {fields && (
+        <>
+          <SelectInput
+            source={`${path}.field`}
+            helperText="style-editor.field-help"
+            style={{ minWidth: '20em', margin: '1em 0' }}
+            label="style-editor.field"
+            choices={fields
+              .filter(field => [1, 2, 3].includes(field.data_type))
```

Comment:
What are `1, 2, 3`? Maybe you should declare theses values by using capital-named const

Reply:
Done